packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.spec.tsx

```diff
@@ -527,5 +528,56 @@ describe("horizontal overflow", (): void => {
         expect(window.removeEventListener).toHaveBeenCalled();
         expect(resizeCallback.mock.calls[0][0]).not.toBe("resize");
     });
+    test("should create a resize observer if it is available", (): void => {
+        const ActualObserver: ConstructableResizeObserver = (window as WindowWithResizeObserver)
+            .ResizeObserver;
+        const construct: jest.Mock<any, any> = jest.fn();
+        // Mock the resize observer
+        class MockObserver {
+            public observe: jest.Mock<any, any> = jest.fn();
+            public unobserve: jest.Mock<any, any> = jest.fn();
+            public disconnect: jest.Mock<any, any> = jest.fn();
+            constructor() {
+                construct();
+            }
+        }
+        (window as WindowWithResizeObserver).ResizeObserver = MockObserver;
+
+        // Render the component
+        const rendered: any = mount(
+            <HorizontalOverflow managedClasses={managedClasses}>
+                {imageSet1}
+            </HorizontalOverflow>
+        );
+
+        expect(construct).toBeCalledTimes(1);
+        // Replace the window to it's original state
+        (window as WindowWithResizeObserver).ResizeObserver = ActualObserver;
+    });
+    test("should disconnect the resize observer when unmounted", (): void => {
+        const ActualObserver: ConstructableResizeObserver = (window as WindowWithResizeObserver)
+            .ResizeObserver;
+        const disconnect: jest.Mock<any, any> = jest.fn();
+        // Mock the resize observer
+        // tslint:disable-next-line:max-classes-per-file
+        class MockObserver {
+            public observe: jest.Mock<any, any> = jest.fn();
+            public unobserve: jest.Mock<any, any> = jest.fn();
+            public disconnect: jest.Mock<any, any> = disconnect;
+        }
```

Comment:
Is there a particular reason you're creating two instances of the class here instead of defining it outside the scope of the tests and reusing?

Reply:
not really, it's just a simple way of keeping track of the correct mock functions and keeping the tests independant. Each class only keeps a reference to the mock functions that it needs